pkg/api/customization/kontainerdriver/formatter.go

```diff
@@ -16,8 +18,31 @@ func Formatter(request *types.APIContext, resource *types.RawResource) {
 			resource.AddAction(request, "activate")
 		}
 	}
-
+	// if cluster driver is a built-in, delete removal link from UI
 	if builtIn, _ := resource.Values[client.KontainerDriverFieldBuiltIn].(bool); builtIn {
 		delete(resource.Links, "remove")
 	}
+	resName := resource.Values[client.KontainerDriverFieldName]
+	// resName will be nil when first added
+	if resName != nil {
+		kontainer, err := clusterHasKontainerDriverAttached(request, resName.(string))
+		if err != nil {
+			logrus.Warnf("error querying cluster schema %v", err)
+		} else if kontainer {
+			// if cluster driver in use, delete removal link from UI
+			delete(resource.Links, "remove")
+		}
+	}
+
+}
+
+func clusterHasKontainerDriverAttached(apiContext *types.APIContext, name string) (bool, error) {
+	clusterSchema := apiContext.Schemas.Schema(&schema.Version, client.ClusterType)
+	clusters, err := clusterSchema.Store.List(apiContext, clusterSchema, &types.QueryOptions{
```

Comment:
We can't find the clusters this way. It is very expensive. We need to use an indexer.